xpartamupp/echelon.py

```diff
@@ -569,14 +580,16 @@ def _iq_board_list_handler(self, iq):
         """Handle incoming leaderboard list requests.
 
         Arguments:
-            iq (sleekxmpp.stanza.iq.IQ): Received IQ stanza
+            iq (IQ): Received IQ stanza
 
         """
         if not iq['from'].resource.startswith('0ad'):
             return
 
         command = iq['boardlist']['command']
-        self.leaderboard.get_or_create_player(sleekxmpp.jid.JID(iq['from'], resource="0ad"))
+        jid = JID(iq['from'])
+        jid.resource = "0ad"
```

Comment:
Would we consider making the form similar to how it's seen previously?

`JID(iq['from']).bare + "/0ad"`

Or maybe higher up we should keep the form where it uses `JID(jid, resource="0ad")`

Reply:
I believe JIDs should be used in the code as instances of the JID class as much as possible, as that provides more appropriate type information and additional functionality compared to strings. That's why I liked the previous notation (`JID(iq['from'], resource="0ad")`) for generating a JID object out of a JID but at the same time replacing its resource. Unfortunately that's not possible anymore with Slixmpp, as the constructor of the JID class there doesn't have a `resource` parameter anymore. That's why I initialized the JID without specifying anything related the resource and set it afterwards. Your suggestion would result in a string instead of a JID object and adjusting it to return a JID would not only result in the need to initialize two JID objects, but also to still do string concatenation (`JID(JID(iq['from']).bare + "/0ad")`).